### PR TITLE
Added CLI content

### PR DESCRIPTION
**Pull request by adityapande-1995**
This PR adds a short description of the proposed CLI tool for service introspection.

### CODE REVIEW

File: rep-2010.rst [hunk 1]
```diff
@@ -234,7 +234,10 @@ Any existing tooling for aiding users in setting up ROS security should consider
 Command line interface
 ----------------------
 
-TODO: Consider proposing an interface for a CLI here. Although it seems a bit tangent to the core feature, I think it is important to the proposal since it makes it so people can immediately start using the feature.
+We can have a tool like ``ros2 service echo example_interfaces/srv/AddTwoInts`` that internally subscribes to the two hidden topics ``example_interfaces/srv/AddTwoInts_RequestEvent`` and
```
Comment (deepanshubansal01) [inline rep-2010.rst]: Just wondering isn't `example_interfaces/srv/AddTwoInts` the service type I think, we would like have the service name instead maybe in `echo`, not sure.

File: rep-2010.rst [hunk 2]
```diff
@@ -234,7 +234,10 @@ Any existing tooling for aiding users in setting up ROS security should consider
...
+``example_interfaces/srv/AddTwoInts_ResponseEvent`` and echoes them. It should also support additional command line parameters used for topics, like ``--filter, --no-arr, --no-str, --include-msg-info``, etc.
```
Comment (deepanshubansal01) [inline rep-2010.rst]: Also, once we echo the service how will we know that whether it is `AddTwoInts_ResponseEvent` or `AddTwoInts_RequestEvent` I suppose their will be a heading after echoing on the command line, just curious.

**Comment by deepanshubansal01** [comment]
I was thinking we could support one more permanent argument other than `echo` in `ros2 service`. 

As a particular service can have multiple clients at a time maybe we can expose an argument for client count or something
`ros2 service client-count` not sure if this makes sense. It will be the equivalent of `hz` in `ros2 topic` but in our case we will just tell how many clients are quering the service at a particular time.

**Comment by deepanshubansal01** [comment]
> I was thinking we could support one more permanent argument other than `echo` in `ros2 service`.
> 
> As a particular service can have multiple clients at a time maybe we can expose an argument for client count or something `ros2 service client-count` not sure if this makes sense. It will be the equivalent of `hz` in `ros2 topic` but in our case we will just tell how many clients are quering the service at a particular time.

Now that I am thinking this is unrelated to the proposed REP we can ignore this ig if it doesn't makes sense.

**Comment by adityapande-1995** [comment]
> > I was thinking we could support one more permanent argument other than `echo` in `ros2 service`.
> > As a particular service can have multiple clients at a time maybe we can expose an argument for client count or something `ros2 service client-count` not sure if this makes sense. It will be the equivalent of `hz` in `ros2 topic` but in our case we will just tell how many clients are quering the service at a particular time.
> 
> Now that I am thinking this is unrelated to the proposed REP we can ignore this ig if it doesn't makes sense.

This is probably very implementation specific, and this REP might not be the best place to include it. We could talk about this in a PR.

File: rep-2010.rst (hunk 1 above)
Comment (adityapande-1995) [reply rep-2010.rst]: Updated

**Review by deepanshubansal01 — APPROVED** [review]

File: rep-2010.rst (hunk 2 above)
Comment (adityapande-1995) [reply rep-2010.rst]: Added an example to clarify. The responses and requests can be separated by the ``request_type`` fields.

File: rep-2010.rst [hunk 3]
```diff
@@ -234,7 +234,46 @@ Any existing tooling for aiding users in setting up ROS security should consider
 Command line interface
 ----------------------
 
-TODO: Consider proposing an interface for a CLI here. Although it seems a bit tangent to the core feature, I think it is important to the proposal since it makes it so people can immediately start using the feature.
+Suppose we have a service named ``/add_two_ints`` of type ``example_interfaces/srv/AddTwoInts``.
+We can have a tool like ``ros2 service echo /add_two_ints`` that internally subscribes to the two hidden topics ``example_interfaces/srv/AddTwoInts_RequestEvent``, ``example_interfaces/srv/AddTwoInts_ResponseEvent`` and echoes them. It should also support additional command line parameters used for topics, like ``--filter, --no-arr, --no-str, --include-msg-info``, etc.
```
Comment (jacobperron) [inline rep-2010.rst]: Since the command `ros2 service` does already exist, I would rephrase this to say we extend the existing tool with an echo verb.

Regarding style, keep the document to one sentence per line. This helps keep future diffs smaller.

File: rep-2010.rst [hunk 4]
```diff
@@ -234,7 +234,46 @@ Any existing tooling for aiding users in setting up ROS security should consider
...
+   -----------------------
+   request_type: RESPONSE_RECEIVED
+   stamp: 1.30
+   client_id: 1234
+   sequence_number: 2
+   request:
+      sum: 3
+   -----------------------
+
+Additional parameters could be included like ``--response-time``, which should print the delay between the time the client sent the request and the server sent the response, or ``--exculde-msg-content`` to just echo the events and not the content of messages.
```
Comment (jacobperron) [inline rep-2010.rst]: "response_time" seems overloaded here, it might be confused with the time a response was received, for example. Maybe "delay" is a better term? Also, I'm not sure what the output would look like in this case. Would delay value be echo'd as part of the response received event only, or something else? IMO, it might be best to be vague here and not mention specific options. These are subject to change anyways over time.

**Review by jacobperron — COMMENTED** [review]
The example looks compelling to me!

I'm wondering if we should move it to the "feature progress" section as something we aspire to implement as part of this REP. I guess the specification section is more about the publishing of the events and not so much the consumers of the events.

File: rep-2010.rst (hunk 4 above)
Comment (adityapande-1995) [reply rep-2010.rst]: Edited the section to remove specific command line options, moved the example to "feature progress" section.

File: rep-2010.rst (hunk 3 above)
Comment (adityapande-1995) [reply rep-2010.rst]: Done

File: rep-2010.rst [hunk 5]
```diff
@@ -234,7 +234,9 @@ Any existing tooling for aiding users in setting up ROS security should consider
 Command line interface
 ----------------------
 
-TODO: Consider proposing an interface for a CLI here. Although it seems a bit tangent to the core feature, I think it is important to the proposal since it makes it so people can immediately start using the feature.
+The existing ``ros2 service`` tool can be extended using an ``echo`` keyword to monitor service events.
+Internally, it would subscribe to the hidden topics and echo them. The existing command line parameters for topics can be extended to be used with this ``echo`` verb, along with new
+arguments on to filter message content and analyse delays.
```
Comment (jacobperron) [inline rep-2010.rst]: style: one sentence per line
```suggestion
Internally, it would subscribe to the hidden topics and echo them.
The existing command line parameters for topics can be extended to be used with this ``echo`` verb, along with new
arguments on to filter message content and analyse delays.
```
